auth/credentials/externalaccount/externalaccount.go

```diff
@@ -0,0 +1,227 @@
+	// not a workload identity pool. The underlying principal must still have
+	// serviceusage.services.use IAM permission to use the project for
+	// billing/quota. Optional.
+	WorkforcePoolUserProject string
+	// UniverseDomain is the default service domain for a given Cloud universe.
+	// This value will be used in the default STS token URL. The default value
+	// is "googleapis.com". It will not be used if TokenURL is set. Optional.
+	UniverseDomain string
+	// TODO(codyoss)
+	// SubjectTokenSupplier is an optional token supplier for OIDC/SAML credentials.
+	// One of SubjectTokenSupplier, AWSSecurityCredentialSupplier or CredentialSource must be provided. Optional.
+	//SubjectTokenSupplier SubjectTokenSupplier
+	// AwsSecurityCredentialsSupplier is an AWS Security Credential supplier for AWS credentials.
+	// One of SubjectTokenSupplier, AWSSecurityCredentialSupplier or CredentialSource must be provided. Optional.
+	//AwsSecurityCredentialsSupplier AwsSecurityCredentialsSupplier
+
+	// Client configures the underlying client used to make network requests
+	// when fetching tokens. Optional.
+	Client *http.Client
+}
+
+type CredentialSource struct {
+	// File is the location for file sourced credentials.
+	// One field amongst File, URL, Executable, or EnvironmentID should be
+	// provided, depending on the kind of credential in question.
+	File string
+	// Url is the URL to call for URL sourced credentials.
+	// One field amongst File, URL, Executable, or EnvironmentID should be
+	// provided, depending on the kind of credential in question.
+	URL string
+	// Executable is the configuration object for executable sourced credentials.
+	// One field amongst File, URL, Executable, or EnvironmentID should be
+	// provided, depending on the kind of credential in question.
+	Executable *ExecutableConfig
+	// EnvironmentID is the EnvironmentID used for AWS sourced credentials.
+	// This should start with "AWS".
+	// One field amongst File, URL, Executable, or EnvironmentID should be provided, depending on the kind of credential in question.
+	EnvironmentID string
+
+	// Headers are the headers to attach to the request for URL sourced
+	// credentials.
+	Headers map[string]string
+	// RegionURL is the metadata URL to retrieve the region from for EC2 AWS
+	// credentials.
+	RegionURL string
+	// RegionalCredVerificationURL is the AWS regional credential verification
+	// URL, will default to `https://sts.{region}.amazonaws.com?Action=GetCallerIdentity&Version=2011-06-15`
+	// if not provided.
+	RegionalCredVerificationURL string
+	// IMDSv2SessionTokenURL is the URL to retrieve the session token when using
+	// IMDSv2 in AWS.
+	IMDSv2SessionTokenURL string
+	// Format is the format type for the subject token. Used for File and URL
+	// sourced credentials.
+	Format *Format
+}
+
+// Format contains information needed to retrieve a subject token for URL or
+// File sourced credentials.
+type Format struct {
```

Comment:
Why are references to Format being changed to pointers?

Reply:
Just for consistency across all structs in this module. We have made all other struct pointer types, even if required. Did not matter before because it was an internal detail, but now it is exposed.